price-feeder/oracle/oracle.go

```diff
@@ -524,6 +531,162 @@ func FilterCandleDeviations(
 	return filteredCandles, nil
 }
 
+// ConvertCandlesToUSD converts any candles which are not quoted in USD
+// to USD by other price feeds.
+func convertCandlesToUSD(
```

Comment:
These candle & ticker specific functions are getting a bit out of hand. I've opened up #936 so that we can clean this up 😄